artifactory/commands/buildinfo/publish.go

```diff
@@ -248,3 +275,187 @@ type partialModule struct {
 	artifacts    map[string]buildinfo.Artifact
 	dependencies map[string]buildinfo.Dependency
 }
+
+// Set build property for each dependency.
+func addDependenciesProps(partialBuildsInfo []*buildinfo.PartialBuildInfo, sm *artifactory.ArtifactoryServicesManager, threads int) error {
+	// Key: dependency sha1, Value: pointer to dependency build property.
+	depBuildMap := make(map[string][]*string)
+	var searchResults []servicesutils.ResultItem
+	// Search dependencies for all partials build info.
+	for _, partialBuildInfo := range partialBuildsInfo {
+		repositoriesSet, err := getLocalRepos(partialBuildInfo.ResolverRepositories, sm)
+		if len(repositoriesSet) == 0 {
+			if err != nil {
+				log.Warn("Fail to find repositories to collect dependencies build-name & build-number", err)
+			}
+			continue
+		}
+		sha1Set := commandutils.NewStringSet()
+		// Map the dependencies sha1 to their Build property.
+		for _, module := range partialBuildInfo.Modules {
+			for i := 0; i < len(module.Dependencies); i++ {
```

Comment:
Can we use the "for each" approach here, instead of using the i int?
This should make the code inside the loop shorter.

Reply:
I'm building here a map of dependencies that points to the 'Build' property for each dependency. Later on, I will update this field.
I can not use the "range" approach because it returns a copy of the original variable